arch/xtensa/include/xtensa_asm2_context.h

```diff
@@ -106,6 +106,18 @@ struct xtensa_irq_base_save_area {
 	uintptr_t fpu15;
 #endif
 
+#if (XCHAL_CP_MASK & (1 << XCHAL_CP_ID_AUDIOENGINELX)) && \
```

Comment:
Can you key this off of core-isa.h values and not the derived junk from the Xtensa HAL?  XCHAL_HAVE_HIFI{3,4,5}, etc...